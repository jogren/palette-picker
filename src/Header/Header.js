import React from 'react';

const Header = ({ generateColors }) => {
  return (
    <header className="Header_header">
      <h1>Palette Picker</h1>
      <button className="generate-color-btn" onClick={generateColors}>Generate New Palette</button>
    </header>
  )
}

export default Header;