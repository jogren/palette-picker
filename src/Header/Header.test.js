import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  let wrapper;
  const generateColorsMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Header 
      generateColors={generateColorsMock}
      lockMessage={true}
      />)
  });

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call generateColors on click of button', () => {
    wrapper.find('button').simulate('click');
    expect(generateColorsMock).toHaveBeenCalled();
  });
});