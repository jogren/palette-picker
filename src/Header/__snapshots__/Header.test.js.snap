// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with the data passed through 1`] = `
<header>
  <h1>
    Palette Picker
  </h1>
  <button
    className="generate-color-btn"
    onClick={[MockFunction]}
  >
    Generate New Palette
  </button>
</header>
`;
